test/v2/services/XmlParsersSpec.scala

```diff
@@ -31,33 +31,81 @@ import scala.xml.NodeSeq
 class XmlParsersSpec extends AnyFreeSpec with TestActorSystem with Matchers with StreamTestHelpers with ScalaFutures with ScalaCheckPropertyChecks {
   "messageTypeExtractor parser" - {
 
-    val validMessageType: NodeSeq =
+    val validDepartureMessageType: NodeSeq =
       <ncts:CC014C PhaseID="NCTS5.0" xmlns:ncts="http://ncts.dgtaxud.ec">
         <HolderOfTheTransitProcedure>
           <identificationNumber>GB1234</identificationNumber>
         </HolderOfTheTransitProcedure>
       </ncts:CC014C>
 
+    val validArrivalMessageType: NodeSeq =
+      <ncts:CC044C PhaseID="NCTS5.0" xmlns:ncts="http://ncts.dgtaxud.ec">
+        <TransitOperation>
+          <MRN>27WF9X1FQ9RCKN0TM3</MRN>
+        </TransitOperation>
+      </ncts:CC044C>
+
+    val invalidDepartureMessageType: NodeSeq =
+      <ncts:CC015C PhaseID="NCTS5.0" xmlns:ncts="http://ncts.dgtaxud.ec">
+        <HolderOfTheTransitProcedure>
+          <identificationNumber>GB1234</identificationNumber>
+        </HolderOfTheTransitProcedure>
+      </ncts:CC015C>
+
+    val invalidArrivalMessageType: NodeSeq =
+      <ncts:CC007C PhaseID="NCTS5.0" xmlns:ncts="http://ncts.dgtaxud.ec">
+        <TransitOperation>
+          <MRN>27WF9X1FQ9RCKN0TM3</MRN>
+        </TransitOperation>
+      </ncts:CC007C>
+
     val invalidMessageType: NodeSeq =
-      <HolderOfTheTransitProcedure>
-        <CC013C>IE013</CC013C>
-      </HolderOfTheTransitProcedure>
+      <TransitOperation>
+          <MRN>27WF9X1FQ9RCKN0TM3</MRN>
+        </TransitOperation>
```

Comment:
I'm curious, why did you change the invalid message type to this? What happens if you leave it as is? `HolderOfTheTransitProcedure` on the root is an invalid message type after all.

Reply:
Mistakenly copy pasted, As added few test cases for arrivals. Reverted back to original.